libs/template/if/src/lib/if.directive.ts

```diff
@@ -549,7 +567,15 @@ export class RxIf<T = unknown>
       this.templateManager.addTemplateRef(RxIfTemplateNames.error, this.error);
     }
     if (changes.rxIf) {
-      this.templateNotifier.next(this.rxIf);
+      if (isSignal(this.rxIf)) {
+        // add to currentRxIfSignal so we can listen to changes
+        this.currentRxIfSignal.set(this.rxIf);
+
+        // send value synchronously
+        this.templateNotifier.next(this.rxIf());
```

Comment:
won't this result in two emissions of `templateNotifier`? Sync + promise coalesced from the effect?

Reply:
Yes, it will emit twice, but I thought that we are already using distinctUntilChanged() in the templateNotifier, so the same value doesn't retrigger change detection.

Reply:
why triggering it in the first place?

Reply:
Don't we want to keep the synchronous behavior? Do we want to always have it scheduled based on the effect scheduler?

Reply:
It would make sense to me to do so, yes